vermouth/processors/locate_charge_dummies.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2018 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+import operator
+import numpy as np
+from .processor import Processor
+
+DEFAULT_DUMMY_ATTRIBUTE = 'charge_dummy'
+
+
+def fibonacci_sphere(n_samples):
+    offset = 2 / n_samples
+    increment = np.pi * (3 - np.sqrt(5))
+    sample_idx = np.arange(n_samples)
+    y = (sample_idx * offset - 1) + offset / 2
+    r = np.sqrt(1 - y * y)
+    phi = (sample_idx % n_samples) * increment
+    x = np.cos(phi) * r
+    z = np.sin(phi) * r
+    return np.stack([x, y, z]).T
+
+
+def colinear_pair():
+    # We want to create a randomly oriented vector of norm 2, and center it on
+    # the origin.
+    vector = np.random.rand(3)
+    vector /= np.linalg.norm(vector)
+    vector *= 2
+    points = np.stack([np.zeros((3, )), vector])
+    points -= points / 2
+    return points
+
+
+def find_anchor(molecule, node_key, attribute_tag=DEFAULT_DUMMY_ATTRIBUTE):
+    dummy = molecule.nodes[node_key]
+    if dummy.get(attribute_tag, None) is None:
+        msg = 'Node "{}" is not a charge dummy. Check the "{}" node attribute.'
+        raise ValueError(msg.format(node_key, attribute_tag))
```

Comment:
A try/except might be better here.

Reply:
I find the if more explicit regarding the logic.